concent_api/utils/tests/test_decorators.py

```diff
@@ -0,0 +1,222 @@
+
+        self.assertEqual(response.status_code, 400)
+        self.assertIn('error', json.loads(response.content))
+
+    def test_handle_errors_and_responses_should_return_http_response_with_serialized_message(self):
+
+        dumped_message = dump(self._create_test_ping_message(), CONCENT_PRIVATE_KEY, self.PROVIDER_PUBLIC_KEY)
+        request = self.request_factory.post("/dummy-url/", content_type = 'application/octet-stream', data = dumped_message)
+
+        response = dummy_view_handle_errors_and_responses(request, dumped_message)
+
+        loaded_response = load(response.content, self.PROVIDER_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIsInstance(loaded_response, message.Ping)
+
+    def test_handle_errors_and_responses_should_return_serialized_message_if_gets_deserialized(self):
+
+        with freeze_time("2017-12-31 00:00:00"):
+            client_auth = message.concents.ClientAuthorization()
+            client_auth.client_public_key = self.PROVIDER_PUBLIC_KEY
+
+        request = self.request_factory.post("/dummy-url/", content_type = 'application/octet-stream')
+
+        with freeze_time("2017-12-31 00:00:10"):
+            response = dummy_view_handle_errors_and_responses(request, client_auth)
+            loaded_response = load(response.content, self.PROVIDER_PRIVATE_KEY, CONCENT_PUBLIC_KEY)  # pylint: disable=no-member
+
+        self.assertEqual(response.status_code, 200)  # pylint: disable=no-member
+        self.assertIsInstance(loaded_response, message.concents.ClientAuthorization)
+        self.assertEqual(loaded_response.timestamp, self._parse_iso_date_to_timestamp("2017-12-31 00:00:00"))
+
+    def test_handle_errors_and_responses_should_return_http_response_if_it_has_been_passed_to_decorator(self):
+
+        request = self.request_factory.post("/dummy-url/", content_type = 'application/octet-stream')
+
+        @handle_errors_and_responses
+        def dummy_view_handle_http_response(_request, _message):
+            http_response = HttpResponse(status = 200)
+            return http_response
+
+        response = dummy_view_handle_http_response(request, self.client_auth)
+
+        self.assertIsInstance(response, HttpResponse)
+        self.assertEqual(response.status_code, 200)
+
+    def test_handle_errors_and_responses_should_return_http_response_if_view_passed_none(self):
+
+        request = self.request_factory.post("/dummy-url/", content_type = 'application/octet-stream')
+
+        @handle_errors_and_responses
+        def dummy_view_handle_none_response(_request, _message):
+            return None
+
+        response = dummy_view_handle_none_response(request, self.client_auth)  # pylint: disable=assignment-from-none
+
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(len(response.content), 0)
+
+    def test_handle_errors_and_responses_should_return_json_response_if_view_passed_dict(self):
```

Comment:
suggestion: you might want to use a bit more sophisticated dicts here - empty dict is a quite specific case;)